drape_frontend/animation/interpolators.cpp

```diff
@@ -214,13 +219,34 @@ ScaleInterpolator::ScaleInterpolator(double delay, double startScale, double end
 // static
 double ScaleInterpolator::GetScaleDuration(double startScale, double endScale, bool isAutoZoom)
 {
-  // Resize 2.0 times should be done for 1.2 seconds in autozoom or for 0.2 seconds in usual case.
-  double const kPixelSpeed = isAutoZoom ? (2.0 / 1.2) : (2.0 / 0.2);
-
   if (startScale > endScale)
     std::swap(startScale, endScale);
 
-  return CalcAnimSpeedDuration(endScale / startScale, kPixelSpeed);
+  double const scaleRatio = endScale / startScale;
+  
+  if (isAutoZoom)
+  {
+    // For auto-zoom, use the original calculation but with a reasonable cap
+    double const kPixelSpeed = 2.0 / 1.2;
+    double duration = CalcAnimSpeedDuration(scaleRatio, kPixelSpeed);
+    // Cap at 1.4s to stay safely under kMaxAnimationTimeSec (1.5s)
+    return std::min(duration, 1.4);
+  }
+  else
+  {
+    // For manual zoom (location button), use a smarter scale-aware duration
+    // Base duration scales with zoom levels but caps reasonably
+    double const logRatio = std::log2(scaleRatio);  // zoom levels difference
+    double const baseDuration = 0.4;
+    double const perZoomLevelDuration = 0.2;
+    double duration = baseDuration + logRatio * perZoomLevelDuration;
+    
+    // Ensure minimum duration for any zoom change
+    duration = std::max(duration, 0.3);
+    
+    // Cap at 1.4s to prevent animation killing while allowing longer comfortable animations
+    return std::min(duration, 1.4);
+  }
```

Comment:
Do we need these changes here and above, since you have implemented the new ParabolicAnimation ?